### PR TITLE
define rules for building and exporting

### LINKED ISSUE #15: define makefile rules for exporting to each target
Currently the exported `Android` apk has all the libraries bunched up in one apk. We should have different apks for each target.

### PR DESCRIPTION
**Pull request by macalimlim**
define rules for building and exporting, resolves #15

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,50 +1,98 @@
+{%- comment -%} dynamic make targets {%- endcomment -%}
+{%- assign target_dir = "./target" -%}
+{%- assign lib_dir = "./native/lib" -%}
+{%- assign bin_dir = "./native/bin" -%}
+{%- assign aarch64-linux-android-debug = "aarch64-linux-android,Android,debug" | split: "|" -%}
+{%- assign aarch64-linux-android-release = "aarch64-linux-android,Android,release" | split: "|" -%}
+{%- assign armv7-linux-androideabi-debug = "armv7-linux-androideabi,Android,debug" | split: "|" -%}
```
Comment (toasteater) [inline Makefile:7]: Not necessarily an issue, but what do the `split: "|"`s here and below accomplish? I don't see any `|`s in the string.

File: Makefile [hunk 2]
```diff
@@ -58,27 +106,8 @@ edit:
...
-
-release-linux-x11: build-x86_64-unknown-linux-gnu-release clean
-	godot --export "Linux/X11" ./native/bin/linux-x11/{{project-name}}.x86_64
-
-release-linux-x11-debug: build-x86_64-unknown-linux-gnu clean
-	godot --export-debug "Linux/X11" ./native/bin/linux-x11/{{project-name}}.debug.x86_64
-
-run: build-x86_64-unknown-linux-gnu
+run:
+	make build-x86_64-unknown-linux-gnu-debug
```
Comment (toasteater) [inline Makefile:110]: Can we make this dependent on the user platform somehow? A lot of users are developing on Windows.

File: Makefile (hunk 1 above)
Comment (macalimlim) [reply Makefile:7]: It puts the string inside an array so I could concat it with the other arrays. The other reason is its becoming long horizontally if I put multiple (or all) targets in one line. There is no way in liquid to initialize arrays...

File: Makefile (hunk 2 above)
Comment (macalimlim) [reply Makefile:110]: That what I was thinking as well, currently `cargo-generate` does not support this, I filed an issue with cargo-generate [here](https://github.com/ashleygwilliams/cargo-generate/issues/251)

Comment (macalimlim) [reply Makefile:110]: my [PR](https://github.com/ashleygwilliams/cargo-generate/pull/252) has been merged, lest just wait for the release. I will create a separate issue for this.

**Review by toasteater — APPROVED** [review]
I'm trusting you that this would work -- the templates here are a little bit too complex for me to understand. We should really get CI working.

**Pull request merged**